htdocs/accountancy/admin/productaccount.php

```diff
@@ -189,7 +189,7 @@
 					$db->begin();
 
 					if (!empty($conf->global->MAIN_PRODUCT_PERENTITY_SHARED)) {
-						$sql = "INSERT INTO ".MAIN_DB_PREFIX."product_perentity (fk_product, entity, '".$db->escape($accountancy_field_name)."')";
+						$sql = "INSERT INTO ".MAIN_DB_PREFIX."product_perentity (fk_product, entity, ".$db->escape($accountancy_field_name).")";
```

Comment:
If you use $db->escape( it means it is a string. In such a cas, the ' ' must be present into the SQL string.
If value is an int, you must use instead   (int) $var  when forging the sql request.